src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +962,1065 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
+            putFieldError(KFSPropertyConstants.REPORTS_TO_CHART_OF_ACCOUNTS_CODE, KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_RPTS_TO_ACCT_REQUIRED_IF_FRINGEBENEFIT_FALSE);
+            result &= false;
+        }
+
+        // if either of the fringe benefit account fields are not present, then we're done
+        if (result == false) {
+            return result;
+        }
+
+        // attempt to load the fringe benefit account
+        Account fringeBenefitAccount = accountService.getByPrimaryId(newAccount.getReportsToChartOfAccountsCode(), newAccount.getReportsToAccountNumber());
+
+        // fringe benefit account must exist
+        if (fringeBenefitAccount == null) {
+            putFieldError(KFSPropertyConstants.REPORTS_TO_ACCOUNT_NUMBER, KFSKeyConstants.ERROR_EXISTENCE, getFieldLabel(Account.class, "reportsToAccountNumber"));
+            return false;
+        }
+
+        // fringe benefit account must be active
+        if (!fringeBenefitAccount.isActive()) {
+            putFieldError(KFSPropertyConstants.REPORTS_TO_ACCOUNT_NUMBER, KFSKeyConstants.ERROR_INACTIVE, getFieldLabel(Account.class, "reportsToAccountNumber"));
+            result &= false;
+        }
+
+        // make sure the fringe benefit account specified is set to fringe benefits = Y
+        if (!fringeBenefitAccount.isAccountsFringesBnftIndicator()) {
+            putFieldError(KFSPropertyConstants.REPORTS_TO_ACCOUNT_NUMBER, KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_RPTS_TO_ACCT_MUST_BE_FLAGGED_FRINGEBENEFIT, fringeBenefitAccount.getChartOfAccountsCode() + "-" + fringeBenefitAccount.getAccountNumber());
+            result &= false;
+        }
+
+        return result;
+    }
+    
+    /**
+     * This method checks to make sure that if the contracts and grants fields are required they are entered correctly
+     *
+     * @param newAccount
+     * @return
+     */
+    protected boolean checkCgRequiredFields(CuAccountGlobal newAccount) {
+        boolean result = true;
+
+        // Certain C&G fields are required if the Account belongs to the CG Fund Group
+        if (ObjectUtils.isNotNull(newAccount.getSubFundGroup())) {
+        	if (getSubFundGroupService().isForContractsAndGrants(newAccount.getSubFundGroup())) {
+        		result &= checkEmptyBOField(KFSPropertyConstants.ACCT_INDIRECT_COST_RCVY_TYPE_CD, newAccount.getAcctIndirectCostRcvyTypeCd(), replaceTokens(KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_ICR_TYPE_CODE_CANNOT_BE_EMPTY));
+	    		 result &= checkEmptyBOField(KFSPropertyConstants.FINANCIAL_ICR_SERIES_IDENTIFIER, newAccount.getFinancialIcrSeriesIdentifier(), replaceTokens(KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_ICR_SERIES_IDENTIFIER_CANNOT_BE_EMPTY));
+	
+	             // Validation for financialIcrSeriesIdentifier
+	             if (checkEmptyBOField(KFSPropertyConstants.FINANCIAL_ICR_SERIES_IDENTIFIER, newAccount.getFinancialIcrSeriesIdentifier(), replaceTokens(KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_ICR_SERIES_IDENTIFIER_CANNOT_BE_EMPTY))) {
```

Comment:
The "checkEmptyBOField" check for financialIcrSeriesIdentifier is being duplicated here. I would recommend refactoring things to avoid the duplicate call, in case it ends up adding the same error message twice.